### PR TITLE
Ws app refactorings

### PR DESCRIPTION
**Pull request by samuelpilz**
Refactor such that the app is more easily tested and the App itself is more independent from Websockets. The goal is that only the Modules in the `app` folder know about Websockets.

### CODE REVIEW

**Comment by power-fungus** [comment]
Note that WsAppUtils can be deleted since there is a typeclass for sending different types

**Comment by fendor** [comment]
Resolve the merge conflicts, then we will merge

File: app/WsConnection.hs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import qualified Data.Aeson         as Aeson
+import qualified Network.WebSockets as WS
+
+
+newtype WsConnection = WsConnection WS.Connection
+
+instance IsConnection WsConnection where
+    type Pending WsConnection = WS.PendingConnection
+
+    sendMsg (WsConnection conn) =
```
Comment (fendor) [inline app/WsConnection.hs]: why msg? they are plain bytestrings, i think `sendData` is a more fitting name.
For me, Message is more high level.

Comment (power-fungus) [inline app/WsConnection.hs]: The type of sendMsg` is in fact `c -> MessageForClient -> IO ()` for better mocking & to be more disconnected from the Websockets in this place

**Comment by power-fungus** [comment]
This makes Testing impure code much easier, closes #2

**Pull request merged**